### PR TITLE
Unlock semver of tchannel and fix back-incompat as-http

### PR DESCRIPTION
**Pull request by kriskowal**
r @anson627 @ShanniLi @malandrew 
cc @jcorbin

### CODE REVIEW

**Comment by anson627** [comment]
thanks for the fix! I didn't realize sendRequest is being used as a public interface.

Ideally we should require tchannel to be passed in from constructor and use it going forward, the send/forward methods should not take tchannel as argument.

**Comment by anson627** [comment]
:+1:

**Comment by jcorbin** [comment]
lgtm

File: as-http.js [hunk 1]
```diff
@@ -61,7 +61,8 @@ TCurlAsHttp.prototype.send = function send() {
         streamed: true,
         hasNoParent: true
     });
-    self.asHttpClient.sendRequest(req, hreq, onSent);
+    self.asHttpClient.channel = self.subChannel.topChannel;
```
Comment (jcorbin) [inline as-http.js]: btw, it's _supposed_ to get it from the req on first use, but we missed that in the "normal" send path; should be done like the forwarding path: https://github.com/uber/tchannel-node/blob/master/as/http.js#L341

**Comment by Raynos** [comment]
Unlocking semver is risky without adding running the `tcurl` tests to travis CI for tchannel.

**Comment by anson627** [comment]
agree, we need link test for tcurl then.

**Pull request merged**